0_json_functions.R

```diff
@@ -348,7 +350,7 @@ export_audit_invalid_data <- function(portfolio_total_, export_path_full) {
   invalidsecurties <- toJSON(portfolio_total_, dataframe = c("columns"))
 
   write(invalidsecurties, file = paste0(export_path_full, ".json"))
-  readr::write_csv(portfolio_total_, path =  paste0(export_path_full, ".csv"))
+  safe_write_csv(portfolio_total_, file = paste0(export_path_full, ".csv"))
```

Comment:
Thanks @cjyetman. How about renaming to `write_csv()` to intentionally overwrite `readr::write_csv()`? That will spread the protection we just built to all calls to `write_csv()`. The only additional change we would need is to remove the namespace from this one call:

![image](https://user-images.githubusercontent.com/5856545/113900403-a1248580-9782-11eb-9790-7cf6905fad86.png)

Reply:
Also, would you like me to add a workflow to check the web-tool with readr < 1.4.0?

Reply:
👏 Don't 👏 intentionally 👏 introduce 👏 namespacing 👏 conflicts 👏

But more seriously, I'd much rather us go through with a regex and replace the function name, and any namespacing, than introduce an identically named function. 

I also think that `safe_write_csv` is a name that promises more than we are planning to put into it right now, but I can't think of a better name off the top of my head. Maybe it'll come to me soon.

@maurolepore Adding checks for readr < 1.4.0 is a good idea, but let's not jump too hard on that yet, since we need to figure out a way to be testing other old packages as well.

Reply:
Okay, I keep a different name. I note that while overwriting well established functions is *generally* a bad idea, it is a powerful way to leverage the power of an interface to inject behavior with o code. Now, for example, if we want to spread the new bahaviour we need to change code in all these places:

![image](https://user-images.githubusercontent.com/5856545/113906828-86094400-9789-11eb-988d-81dc21a649f4.png)

--

Yeah, I agree that "safe" is vague. How about `write_csv_file()`? It reads well and the `_file` bit hints about the `file` argument. 

--

We have a CI pipeline that runs the web tool on gh actions. It's doesn't exactly use the production environment (we could) but it's close. My idea would be to copy the workflow or extend the existing one to run the web tool after a call to `remotes::install_version("readr", "1.3.1")`.

Reply:
I'm very hesitant with overwriting function names like that.... then it's very easy for a function to have different behavior than you expect... it might cause confusion with the loading order... etc.

Reply:
> Also, would you like me to add a workflow to check the web-tool with readr < 1.4.0?

I'm not opposed to that, but honestly I'm still struggling to understand how these workflow.md things are meant to be used, so... if you feel there's value in that, then go for it.

Reply:
Oh, the name is confusing. The workflow I'm now talking about is a gh-actions workflow in .github/workflows/ to test the web tool with R < 1.4.0 on CI. 

The file workflow.md has nothing to do with CI or this PR -- it's meant to document the process and tools that people need to run PACTA.

Reply:
thanks for the clarification, I misunderstood